NextcloudTalk/FileMessageTableViewCell.m

```diff
@@ -206,24 +218,43 @@ - (void)setupForMessage:(NCChatMessage *)message withLastCommonReadMessage:(NSIn
+                                           
+                                           CGFloat previewHeight = isMediaFile ? kFileMessageCellMediaFilePreviewHeight : kFileMessageCellFilePreviewHeight;
+                                           CGFloat previewMaxWidth = isMediaFile ? kFileMessageCellMediaFileMaxPreviewWidth : kFileMessageCellFileMaxPreviewWidth;
+                                           
+                                           if (height > previewHeight) {
+                                               CGFloat ratio = previewHeight / height;
                                                width = width * ratio;
-                                               height = kFileMessageCellFilePreviewHeight;
+                                               height = previewHeight;
                                            }
-                                           if (width > maxPreviewImageWidth) {
-                                               CGFloat ratio = maxPreviewImageWidth / width;
-                                               width = maxPreviewImageWidth;
+                                           if (width > previewMaxWidth) {
+                                               CGFloat ratio = previewMaxWidth / width;
+                                               width = previewMaxWidth;
                                                height = height * ratio;
                                            }
                                            weakSelf.vPreviewSize[3].constant = height;
                                            weakSelf.hPreviewSize[3].constant = width;
                                            weakSelf.vGroupedPreviewSize[1].constant = height;
                                            weakSelf.hGroupedPreviewSize[1].constant = width;
+                                           // if the image is very narrow, use a very small corner radius
+                                           if (height < 2 * kFileMessageCellMediaFilePreviewCornerRadius || width < 2 * kFileMessageCellMediaFilePreviewCornerRadius) {
+                                               weakSelf.previewImageView.layer.cornerRadius = MIN(height, width) / 2.0;
+                                           }
+                                           // use a bigger corner radius for media file previews since they're shown bigger
+                                           else {
+                                               weakSelf.previewImageView.layer.cornerRadius = isMediaFile ?  kFileMessageCellMediaFilePreviewCornerRadius : kFileMessageCellFilePreviewCornerRadius;
+                                           }
+                                           // only show the play icon if there is an image preview (not on top of the default video placeholder)
+                                           weakSelf.playIconImageView.hidden = !isVideoFile;
```

Comment:
You could move that inside the following `if` and just do `.hidden = NO`, because you reset it in the cell reuse, you don't need to set it to `YES` here.